src/components/smartLink.tsx

```diff
@@ -1,3 +1,6 @@
+'use client';
```

Comment:
why do we need `'use client';`?

Reply:
In NextJs components are server components per default and therefore are unable to implement any interactivity like `onClick` callbacks. The `'use client'` directive tells React to "make them interactive" once the html is rendered in the browser.